doc/website/blog/tmp-versioned-migrations-sum-file.md

```diff
@@ -0,0 +1,249 @@
+
+```
+
+```sql title="20220504114411_initial.down.sql"
+-- reverse: create "users" table
+DROP TABLE `users`;
+
+```
+
+```text title="atlas.sum"
+h1:SxbWjP6gufiBpBjOVtFXgXy7q3pq1X11XYUxvT4ErxM=
+20220504114411_initial.down.sql h1:OllnelRaqecTrPbd2YpDbBEymCpY/l6ihbyd/tVDgeY=
+20220504114411_initial.up.sql h1:o/6yOczGSNYQLlvALEU9lK2/L6/ws65FrHJkEk/tjBk=
+
+```
+
+As you can see the `atlas.sum` file contains one entry for each migration file generated. With the `atlas.sum`
+generation file enabled, both Team A and Team B will have such a file once they generate migrations for a schema change.
+Now the version control will raise a merge conflict once the second Team attempts to merge their feature. 
+
+![atlas-versioned-migrations-no-conflict](https://entgo.io/images/assets/migrate/conflict.svg)
+
+:::note
+You need to have the Atlas CLI installed on your system in order to follow along the next steps in this post. If you
+haven't yet, please follow the installation
+instructions [here](https://atlasgo.io/cli/getting-started/setting-up#install-the-cli).
+:::
+
+You can check at any time, if your `atlas.sum` file is in sync with the migration directory with the following command (
+which should not output any errors now):
+
+```shell
+atlas migrate validate
+```
+
+However, if you'd happen to make a manual change to your migration files, like adding a new SQL statement, editing an
+existing one or even creating a complete new file, the `atlas.sum` file is no longer in sync with the migration
+directories contents. Attempting to generate new migration files for a schema change will now be blocked by the Atlas
+migration engine. Try it out by creating a new empty migration file and run the `main.go` once again:
+
+```shell
+atlas migrate new migrations/manual_version.sql --format golang-migrate
+go run -mod=mod main.go initial
+# 2022/05/04 15:08:09 failed creating schema resources: validating migration directory: checksum mismatch
+# exit status 1
+
+```
+
+The `atlas migrate validate` command will tell you the same:
+
+```shell
+atlas migrate validate
+# Error: checksum mismatch
+# 
+# You have a checksum error in your migration directory.
+# This happens if you manually create or edit a migration file.
+# Please check your migration files and run
+# 
+# 'atlas migrate hash --force'
+# 
+# to re-hash the contents and resolve the error.
+# 
+# exit status 1
+```
+
+In order to get the `atlas.sum` file back in sync with the migration directory, we can once again use the Atlas CLI:
+
+```shell
+atlas migrate hash --force
+```
+
+As a safety measure, the Atlas CLI does not operate on a migration directory, that is not in sync with its `atlas.sum`
+file. Therefore, you need to add the `--force` flag to the command. 
+
+For cases, where a developer forgets to update the `atlas.sum` file after making a manual change, you can add
+a `atlas migrate validate` call to your CI.
```

Comment:
Either provide an example or let's mention here that we plan to provide a GitHub action example for it in the next blog post.